### DISCUSSION TITLE
Just some Questions (SV06-Plus)

### DISCUSSION THREAD

**Discussion by asusdk** [bassamanator/Sovol-SV06-firmware#28] [Q&A]
Hi there ^^

at first let me thank you @bassamanator for you're very useful project and the hard work that It's linked with.

I have now 4 Years of experience with 3D-Printing, sadly on a, from the beginning bad working Ender3-Pro. But at least, I learned a lot in these Years, and if you read a lot, and watch a ton of Youtube-Videos about this Topic, you hear some "general-knowledge". 

As everyone tells everywhere, I thought I should avoid raising the Microsteps above 16, otherwise I will drop accuracy.
I did not had the chance to test your config yet, because I still wait for the shipment of my sovol sv06 plus.

So, I would like to ask why you have set the Microsteps to 128 ? Just to fix some of the VFA´s ? Does it maybe NOT lower Accuracy ? I'm not sure what to think/believe right now, so perhaps you could give me some hints, or explain the thought behind this decision ?

Also, if I would like to try your config, but lower the Microsteps, did I need to modify the Rotationdistance as well ? Or just set a lower value for the Microsteps itself  ?

I did not find the use of Firmware retraction in this profile, and I used it now for a few years, so I would like to know if it's better not to use it ? Or do you just prefer to set it in your slicer on a per-filament-basis ?

And last but not least, how to use the TEST_SPEED Macro, does it tell me which speeds I should use ? And when yes, how to use the Measurements ? Or does this Macro Calibrates something automatically ?

Thanks a lot in advance =)

**Comment by bassamanator** [comment]
Please look here to find out about how to enable [firmware retraction](https://www.klipper3d.org/G-Codes.html?h=fir#firmware_retraction). I have never found a need for it. What I do is to set retraction length in the printer settings in SuperSlicer. If a particular filament needs a different retraction length, or speed, I make those adjustments in the `overrides` section of that filament's settings.

The `TEST_SPEED` macro is not imported in the `printer.cfg`, and there is a warning in the file that states that it is not ready for use. It is a brilliant macro that works beautifully on my Voron 0.1, however, I have not gotten it to work properly with my SV06. The author of the macro has a guide [here](https://ellis3dp.com/Print-Tuning-Guide/articles/determining_max_speeds_accels.html). Feel free to `include` it at the top of your `printer.cfg` and test it out, however, be ready to do an emergency stop. You have been warned. If you get this working, please let me know, so I can include it in this Klipper configuration officially.

`microsteps` in the `printer.cfg` is a 'stand-alone' settings, so feel free to adjust it to your needs, and no other changes will be required.

Please provide a **well documented** source that states that TMC2209 drivers lose positional accuracy with higher microsteps, I assure you that you will not find one. These drivers can microstep all the way up to 256, which I have tested for many hours. Unfortunately, the mcu on this board cannot reliably generate those microsteps under all speed and movement conditions, that's why I've reduced them to 128. I'm curious, what do you think will help you in achieving positional accuracy or extrusion accuracy: a motor that can divide a full rotation by 1/16, or one that can divide a full rotation by 1/256. It makes sense that VFAs are greatly reduced as you increase microsteps, it's because you've increased positional accuracy of the rotation.

My first and foremost concern with any printer is print accuracy and print quality, nothing is more important than that. That has been my philosophy in developing this Klipper configuration. You can see some random prints [here](https://github.com/bassamanator/everything-sovol-sv06/blob/main/images/prints/README.md).

Thank you for the questions.

**Reply by asusdk** [reply]
Thanks a lot for this detailed explanation, i cant wait to test this config =)

**Reply by bassamanator** [reply]
I'll try to create a post about this, or perhaps just pin this discussion.

Be sure to familiarize your self with SuperSlicer. The [AndrewEllis](https://ellis3dp.com/Print-Tuning-Guide/) print settings, which are included on this repo, are second to none!

All the best!

**Reply by asusdk** [reply]
For sure, I used Cura most of the time, but also I used Superslicer, I've already installed it, and imported the profile and also corrected for the bigger printspace of the plus model. I'm just sometimes not sure if I like it or not, because there are so few updates, and I don't even know if the latest stable release already contains Arachne or not, so I will try it for sure.

**Reply by bassamanator** [reply]
Arachne is not in the latest official release as far as I know.

You know you're not the first person to mention updates. I would love to know what special designs people are printing, parts that require the latest innovation from the slicer world.

As an experiment, I would love to compare parts printed using SS vs Cura or PS let's say. Don't get me wrong, I'm expecting to see a difference because the Arachne engine should produce greater details.

**Reply by asusdk** [reply]
Hm I think it's more a combination of things I prefer at cura, not related to the print quality I would say I prefer: the interface, that I can fine-tune literally everything. And print related, I like the Z-Seam-Placement of Cura way better, also that i can control some of the movement, like always or never move along already printed parts. The next thing is, that i not 100% trust, SS because that's not a "official Team" like the Guys from Cura or Prusa Slicer, the Infill Options are better, and so on, sometimes it's in the small Details.

**Reply by bassamanator** [reply]
I hear you. I'm a huge Cura fan actually, and it took me a long time to make the switch to SS. I can't go back though. First because print quality is too important to me, which means I need print settings that have been created by a maniac such as myself. Second, I spend so little time touching anything in SS, whereas in Cura it would take me forever to finally get a print started; this applies more to people who are changing filament all the time perhaps.

Anyway, if you get a nice profile working in Cura, please consider sharing it here, and I can include it as part of the package on this repo. You'll get all the credit for this of course.
